### ISSUE TITLE
Add MaxViT blocks for interior of model

### ISSUE THREAD

**Issue by Raahul-Singh** [openclimatefix/metnet#56]
For reference:
- [The original paper](https://arxiv.org/abs/2204.01697)
- [Official TF implementation](https://github.com/google-research/maxvit)
- [Unofficial Torch implementation](https://github.com/ChristophReich1996/MaxViT)
- [timm implementation](https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/maxxvit.py)

From the paper:
- All convolutions have kernels of size (3, 3) and are not dilated.
- normalized keys and queries are used
- All attention windows have size 8 by 8
- 32 attention heads
- skip connections from the output of each MaxVit block to the output of MaxVit. More precisely, the final output of MaxVit is a linear transformation of the outputs after each sub-block (after summing with the residual branch)
- MBConv in MaxVit uses the expansion rate of 4
- squeeze-and-excitation in the MBConv have bottleneck ratio of 0.25.
- pre-activation layer normalization throughout the network. 
- layer normalization after each convolution which is not the last convolution in the given sub-block.
- GELU inside MBConv (in MaxVit) and ReLu in all other places.
- stochastic depth in MaxVit with the probability of dropping a given sub-module (i.e. MBConv, local attention or gridded attention) increasing linearly thorough the network from 0 to 0.2. 

Tasks:
- [x] Add MBConv Layers
   - [x] Add skeleton code
   - [x] Add downsampling support
- [x] Add Grid Attention  Layers
- [x] Add Block Attention Layers 
- [x] Make metnet3 specific modifications
- [x] Add documentation
- [x] Test the dims and shapes
- [x] Figure out how to handle the configs

**Comment by Raahul-Singh** [comment]
Quoting from the MetNet3 paper (see : [Supplement: Model and Training](https://arxiv.org/pdf/2306.06079v2.pdf) (page 17))

>We use 12 modified MaxVit [16] blocks. We introduced the following modifications compared to
the original architecture: we removed MLP sub-blocks which were present in the original MaxVit architecture,
we use normalized keys and queries [3] and we introduce skip connections from the output of each MaxVit
block to the output of MaxVit. More precisely, the final output of MaxVit is a linear transformation of
the outputs after each sub-block (after summing with the residual branch). All attention windows have
size 8 by 8 and we use 32 attention heads. MBConv [7] in MaxVit uses the expansion rate of 4, and
squeeze-and-excitation (SE, [8]) with the bottleneck ratio of 0.25.

[timm's implementation](https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/maxxvit.py) has the basic blocks of MaxViT implemented. We can use these and make the aforementioned modifications.
